src/execution/parsing/parse_utils.c

```diff
@@ -15,3 +16,35 @@ const char *ScriptCommand_GetFunctionName(RedisModuleString *functionName) {
     const char *functionName_cstr = RedisModule_StringPtrLen(functionName, NULL);
     return functionName_cstr;
 }
+
+int ParseKeysArgs(RedisModuleCtx *ctx, RedisModuleString **argv, int argc, RAI_Error *err) {
+    if (argc < 3) {
+        RAI_SetError(err, RAI_EDAGBUILDER, "ERR Missing arguments after KEYS keyword");
+        return -1;
+    }
+
+    long long n_keys;
+    const int retval = RedisModule_StringToLongLong(argv[1], &n_keys);
+    if (retval != REDISMODULE_OK || n_keys <= 0) {
+        RAI_SetError(err, RAI_EDAGBUILDER, "ERR Invalid or negative value found in number of KEYS");
+        return -1;
+    }
+
+    // Go over the given args and verify that these keys are located in the local shard.
+    int key_pos = 0;
+    for (size_t argpos = 2; (argpos < argc) && (key_pos < n_keys); argpos++) {
+        if (!VerifyKeyInThisShard(ctx, argv[argpos])) {
+            RAI_SetError(err, RAI_EDAGBUILDER, "ERR CROSSSLOT Keys don't hash to the same slot");
```

Comment:
consider adding `RAI_Error` to `VerifyKeyInThisShard`
also, consider rephrase the error message so we can differentiate between Redis error and RedisAI error